boa3/internal/analyser/analyser.py

```diff
@@ -58,7 +60,7 @@ def __init__(self, ast_tree: ast.AST, path: str = None, project_root: str = None
     def analyse(path: str, log: bool = False,
                 imported_files: Optional[Dict[str, Analyser]] = None,
                 import_stack: Optional[List[str]] = None,
-                root: str = None) -> Analyser:
+                root: str = None, env: str = None, **kwargs) -> Analyser:
```

Comment:
Why use `**kwargs` if you know what parameters you want? In this case, you only use it to check for a `compiler_entry` arg.

Wouldn't it be better to just explicitly to say it, like: 
```python
    def analyse(path: str, log: bool = False,
                imported_files: Optional[Dict[str, Analyser]] = None,
                import_stack: Optional[List[str]] = None,
                root: str = None, env: str = None, compiler_entry: bool =False) -> Analyser:
```
you could even use the `*` to indicate that this parameter is different from the others and with the help of the comment explain why:
``` python
    def analyse(path: str, log: bool = False,
                imported_files: Optional[Dict[str, Analyser]] = None,
                import_stack: Optional[List[str]] = None,
                root: str = None, env: str = None, *, compiler_entry: bool =False) -> Analyser:
```